canonical/ubuntu.com
# WD-4664 Certified filters

## Done

- Dynamically load filters on /certified pages

## QA

- Go to https://ubuntu-com-13088.demos.haus/certified?q=nvidia 
  - Click on more than 1 category filter
  - Select vendor and release
  - Click apply, check that it works and results include the filters you selected.
  - Check server, click on "Show all vendors", uncheck server and click on "Show all vendors", the new filters for vendors should update accordingly.

- Go to https://ubuntu-com-13088.demos.haus/certified?q=g5 
  - Click on only one category filter
  - Select vendor and release
  - Click apply, check that it goes to the selected `certified/<category>` page

- Go to https://ubuntu-com-13088.demos.haus/certified/laptops and click on the "Desktop" category filter
It should take you to the general search page.

- Go to https://ubuntu-com-13088.demos.haus/certified?q=nvidia&category=Laptop&category=Server&category=Ubuntu+Core should show the filters loaded

- Do the first two steps without vendor and release filters
- Do the first two steps without vendor selected.
- Do the first two steps without releases selected.

- Click on "Show more" and "Show less"

Some more edge cases and UX requirements:
- Go to https://ubuntu-com-13088.demos.haus/certified?q=nvidia&limit=20 click on SoC filter and apply, it should not go 404 now, it should redirect correctly to the SoC page.

- https://ubuntu-com-13088.demos.haus/certified?q=nvidia&category=Laptop&category=Desktop click on SoC, then click on "Show all vendors", it should show the filters.

- https://ubuntu-com-13088.demos.haus/certified/laptops click on "Desktop", listed vendors should match the list of Laptops + Desktop's selected filters.

- Go to https://ubuntu-com-13088.demos.haus/certified?q=nvidia&category=Laptop&category=Server&category=Ubuntu+Core, vendors and release filters should load

- https://ubuntu-com-13088.demos.haus/certified/laptops?q=nvidia&limit=20&category=Laptop&category=Desktop should redirect to the [generic search page](https://ubuntu-com-13088.demos.haus/certified?q=) with filters selected.

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-4664?atlOrigin=eyJpIjoiODRiNDA0ODdlZWI4NDExNTg4NDBlZDczNTRhYjkxM2QiLCJwIjoiaiJ9

## Screenshots

[If relevant, please include a screenshot.]


## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)
